server/src/main/java/org/opensearch/index/fielddata/fieldcomparator/UnsignedLongValuesComparatorSource.java

```diff
@@ -81,9 +81,17 @@ public Object missingObject(Object missingValue, boolean reversed) {
             return min ? Numbers.MIN_UNSIGNED_LONG_VALUE : Numbers.MAX_UNSIGNED_LONG_VALUE;
         } else {
             if (missingValue instanceof Number) {
-                return ((Number) missingValue);
+                long ul = ((Number) missingValue).longValue();
+                if (ul < 0) {
+                    throw new IllegalArgumentException("missing value of type [unsigned_long] must not be negative");
+                }
+                return BigInteger.valueOf(ul);
             } else {
-                return new BigInteger(missingValue.toString());
+                BigInteger missing = new BigInteger(missingValue.toString());
+                if (missing.signum() < 0) {
+                    throw new IllegalArgumentException("missing value of type [unsigned_long] must not be negative");
```

Comment:
For consistency with other error reports:

```suggestion
                    throw new IllegalArgumentException("Value [" + missingValue + "] is out of range for an unsigned long");
```